gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+xor2() {
+	echo -n "${1:0:1}"
+	echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
+}
+
+interface_linklocal() {
+	# We generate a predictable v6 address
+	local macaddr="$(echo $(uci get wireguard.wireguard.privatekey | wg pubkey) |md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
+	local oldIFS="$IFS"; IFS=':'; set -- $macaddr; IFS="$oldIFS"
+	echo "fe80::$(xor2 "$1")$2:$3ff:fe$4:$5$6"
+}
+
+clean_port(){
+	echo "$(echo $1 | sed -r 's/:[0-9]+$|\[|\]//g')"
+}
+
+check_handshake() {
+	local MESH_VPN_IFACE=$(uci get wireguard.wireguard.iface)
+	wg show $MESH_VPN_IFACE | grep -q handshake
+	if [ $? == 1 ]; then
+		local gateway="$(clean_port $1)"
+		local peer="$(clean_port "$(wg show $MESH_VPN_IFACE | grep endpoint | awk '{print $2}')")"
+		local other_ip="$(gluon-wan nslookup $gateway | grep -v $peer | grep 'Address [0-9]')"
+		if [ "$other_ip" == "" ]; then
+			exit 1
+		fi
+		echo "$(echo $other_ip | sed -r s/Address\ [0-9]://)$(echo $1 | egrep -oe :[0-9]+$)"
+		exit 0
+	fi
+	exit 1
+}
+
+# Do we already have a private-key? If not generate one
+temp=$(uci get wireguard.wireguard.privatekey);
+if [ "$?" -ne "0" ]; then
+	uci set wireguard.wireguard.privatekey=$(wg genkey);
+	uci commit wireguard
+fi
+
+# Is wireguard enabled?
+if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wireguard.wireguard.enabled)" == "1" ]; then
+
+	#We assume we are not connected by default
+	CONNECTED=0
+
+	MESH_VPN_IFACE=$(uci get wireguard.wireguard.iface)
+
+	# Check connectivity to supernode
+	ping -c1 -w2  $(wg  | grep fe80 | cut -d'/' -f1 | awk '{print $3}')%$MESH_VPN_IFACE
```

Comment:
Not sure what you actually want here, but `wg show $MESH_VPN_IFACE endpoints` or `wg show $MESH_VPN_IFACE allowed-ips` might be better for being machine parsable.